elixir/apps/domain/priv/repo/migrations/20250512171004_backfill_missing_emails.exs

```diff
@@ -0,0 +1,18 @@
+defmodule Domain.Repo.Migrations.BackfillMissingEmails do
+  use Ecto.Migration
+
+  def up do
+    execute("""
+      UPDATE auth_identities ai
+      SET email = ai.provider_identifier
+      FROM auth_providers ap
+      WHERE ai.provider_id = ap.id
+      AND ap.adapter = 'email'
+      AND ai.email IS NULL;
+    """)
```

Comment:
Would it make sense to have a partial unique index here enforcing this?

Reply:
What type of uniqueness are you thinking here?  

The only thing that comes to mind is:
```
(account_id, adapter_type, email) WHERE adapter_type = 'email'
```

But if that's the case, it would probably be easier to add a `NOT NULL WHERE adapter_type = 'email'` constraint wouldn't it?

Reply:
Yeah sorry I was thinking something similar. Just to avoid the data getting into weird states that could fail a future migration as we work through this stuff.

Reply:
Ah I see. Yeah we can forget it then.